aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -672,7 +724,7 @@ func resourceAviatrixSpokeGatewayRead(d *schema.ResourceData, meta interface{})
 			d.Set("vpc_id", strings.Split(gw.VpcID, "~~")[0]) //AWS vpc_id returns as <vpc_id>~~<other vpc info> in rest api
 			d.Set("vpc_reg", gw.VpcRegion)                    //AWS vpc_reg returns as vpc_region in rest api
 
-			if gw.AllocateNewEipRead {
+			if gw.AllocateNewEipRead && !gw.EnablePrivateOob {
```

Comment:
If `enable_private_oob == true` then `allocate_new_eip` will always be read as false. However,  `allocate_new_eip` is default `true`. Thus if someone does not explicitly set `allocate_new_eip` to false then they would have a diff after apply. To me this implies that setting `enable_private_oob == true && allocate_new_eip == true` is an invalid config and we should throw an error if that occurs.

Reply:
Now I use the following `DiffSuppressFunc` for `allocate_new_eip`. The idea is to completely ignore `allocate_new_eip` when oob is enabled.

```
DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
	return d.Get("enable_private_oob").(bool)
},
```

Reply:
Ohhhh I see now, got it.